clippy_lints/src/loops/same_item_push.rs

```diff
@@ -19,19 +20,30 @@ pub(super) fn check<'tcx>(
     _: &'tcx Expr<'_>,
     body: &'tcx Expr<'_>,
     _: &'tcx Expr<'_>,
+    msrv: &Msrv,
 ) {
-    fn emit_lint(cx: &LateContext<'_>, vec: &Expr<'_>, pushed_item: &Expr<'_>, ctxt: SyntaxContext) {
+    fn emit_lint(cx: &LateContext<'_>, vec: &Expr<'_>, pushed_item: &Expr<'_>, ctxt: SyntaxContext, msrv: &Msrv) {
         let mut app = Applicability::Unspecified;
         let vec_str = snippet_with_context(cx, vec.span, ctxt, "", &mut app).0;
         let item_str = snippet_with_context(cx, pushed_item.span, ctxt, "", &mut app).0;
 
+        let help = if msrv.meets(msrvs::REPEAT_N)
+            && let Some(std_or_core) = std_or_core(cx)
+        {
+            format!(
+                "consider using vec![{item_str};SIZE] or {vec_str}.extend({std_or_core}::iter::repeat_n({item_str}, SIZE))"
+            )
+        } else {
+            format!("consider using vec![{item_str};SIZE] or {vec_str}.resize(NEW_SIZE, {item_str})")
```

Comment:
Optional suggestion:

I'm guessing that the help message might be a bit easier if we split it up into two messages. Especially if the `item_str` is more complicated. I'm imagining something like this:

```
   = help: consider using vec![item;SIZE]
   = help: or resize(NEW_SIZE, item)
```

You can use `span_lint_and_then` for this. The first help message can always be the same and just the second one depends on the MSRV.

I'm a little bit surprised that this isn't emitted as a suggestion. But that's just an observation you can ignore as well.

Reply:
It is clearer with multi lines indeed.

Concerning the suggestion, it would be hard to do so: not only can the loop contain more than the push statement, but also it can depend on items defined inside the loop (but constant), such as in:

```rust
for _ in a..b {
  let item = 2;
  vec.push(item);
}
```

Also, if `b-a` can be negative, or not of the right type, then we wouldn't be able to use them as suggested.